soc/arm/nxp_imx/rt/arm_mpu_regions.c

```diff
@@ -20,7 +20,14 @@ static const struct arm_mpu_region mpu_regions[] = {
 	/* Region 1 */
 	MPU_REGION_ENTRY("SRAM_0",
 			 CONFIG_SRAM_BASE_ADDRESS,
+#ifdef CONFIG_USB
+			 { \
+				(NORMAL_OUTER_INNER_NON_CACHEABLE_NON_SHAREABLE | \
+				 MPU_RASR_XN_Msk | REGION_SRAM_0_SIZE | P_RW_U_NA_Msk) \
+			 }),
+#else
```

Comment:
@MaureenHelm USB examples need disable the cacheable of RAM. I cannot find the better way to change the RAM cacheable attribute, I do as the above changes. could you please help review the changes?

Reply:
@MarkWangChinese You probably need to enable NOCACHE support and define the area you want to be non-cacheable with __nocache. Look for example what is done in the Ethernet MCUX driver.

Reply:
@MarkWangChinese you need the following to use cache and nocache in your usb driver: 
1. add USB_DEVICE_CONFIG_BUFFER_PROPERTY_CACHEABLE just like FSL_SDK_ENABLE_DRIVER_CACHE_CONTROL
2. select NOCACHE_MEMORY and ARM_MPU when configuring USB_DC_NXP_EHCI, just like ETH_MCUX's menuconfig
3. #define USB_CONTROLLER_DATA __nocache similar to SDK examples
4. ep_buf_pool set as __nocache: __nocache K_MEM_POOL_DEFINE(ep_buf_pool, 16, 1024, EP_BUF_NUMOF_BLOCKS, 4); this looks strange but the definition works.

Reply:
@agansari @aurel32  Thanks, I have updated it, you can review it.